src/uu/ls/src/ls.rs

```diff
@@ -1815,6 +1849,17 @@ fn display_file_name(path: &PathData, config: &Config) -> Option<Cell> {
         }
     }
 
+    if config.format != Format::Long && config.blocks {
+        if let Some(md) = path.md() {
+            width += max_blocks + 1;
+            name.insert_str(0, &format!("{} ", pad_left(get_display_size(md).to_string(), max_blocks)));
```

Comment:
Maybe we can make `pad_left` generic over types implementing `Display`? I think it also makes sense to put the `width += max_blocks + 1;` outside of the `if`-`else`.